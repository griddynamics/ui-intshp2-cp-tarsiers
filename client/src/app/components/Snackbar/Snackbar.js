import React, { PureComponent } from 'react';
import './Snackbar.scss';

export default class Snackbar extends PureComponent {
  message = '';

  state = { isActive: false };

  openSnackBar = (message = 'Something went wrong...') => {
    this.message = message;
    this.setState({ isActive: true }, () => {
      setTimeout(() => {
        this.setState({ isActive: false });
      }, 3000);
    });
  };

  render() {
    const { isActive } = this.state;

    return (
      <div className={isActive ? 'snackbar show' : 'snackbar'}>
        {this.message}
      </div>
    );
  }
}
