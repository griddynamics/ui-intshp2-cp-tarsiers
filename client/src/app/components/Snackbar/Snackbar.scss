@import '../../../styles/variables';
@import '../../../styles/mixin';

.snackbar {
  visibility: hidden;
  min-width: 250px;
  margin-left: -125px;
  background-color: $color-dark;
  color: $color-white;
  text-align: center;
  border-radius: 2px;
  padding: 16px;
  position: fixed;
  z-index: 1;
  left: 50%;
  bottom: 5px;
  font-size: 1rem;
  &.show {
    visibility: visible;
    -webkit-animation: fadeIn 0.5s, fadeOut 0.5s 3.5s;
    animation: fadeIn 0.5s, fadeOut 0.5s 3.5s;
  }
}
