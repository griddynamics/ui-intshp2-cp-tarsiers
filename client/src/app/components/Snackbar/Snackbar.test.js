import React from 'react';
import { shallow } from 'enzyme';
import Snackbar from './Snackbar';

describe('<Snackbar />', () => {
  const props = { isActive: false };

  it('should match its snapshot', () => {
    const wrapper = shallow(<Snackbar {...props} />);

    expect(wrapper).toMatchSnapshot();
  });

  setTimeout(() => {
    it('should check state status', () => {
      const wrapper = shallow(<Snackbar {...props} />);

      expect(wrapper.setState('isActive')).toBe(false);
    }, 3000);
  });
});
