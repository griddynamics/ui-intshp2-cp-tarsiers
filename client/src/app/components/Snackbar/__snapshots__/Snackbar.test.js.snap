// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Snackbar /> should match its snapshot 1`] = `
<div
  className="snackbar"
/>
`;
